src/components/Button/Button.tsx

```diff
@@ -14,101 +14,105 @@ import styles from './Button.module.scss';
-   */
-  variant?: ButtonVariant;
-  /**
-   * Additional props to be spread to rendered element
-   */
-  [x: string]: any; // eslint-disable-line
+
+interface BaseButtonProps {
+     /**
+      * Contents of the button.
+      */
+     children?: ReactNode;
+     /**
+      * Additional ClassNames to add to button.
+      */
+     className?: string;
+     /**
+      * Button takes up the full width of its parent container.
+      */
+     fullWidth?: boolean;
+     /**
+      * Name of the icon to include before the button text
+      */
+     iconPrefix?: IconName;
+     /**
+      * Name of the icon to include after the button text
+      */
+     iconSuffix?: IconName;
+     /**
+      * A unique identifier for the button.
+      */
+     id?: string;
+     /**
+      * URL to navigate to when clicked. Passing this attribute automatically
+      * renders an anchor <a> tag, NOT a <button> element.
+      */
+     href?: string;
+     /**
+      * Disables the button, making it inoperable.
+      */
+     isDisabled?: boolean;
+     /**
+      * Replaces the button text with a loading indicator and disables the button.
+      */
+     isLoading?: boolean;
+     /**
+      * Renders an outline version of the button. With a transparent background.
+      */
+     isOutlined?: boolean;
+     /**
+      * Render the button with no styles of its own, other than a pointer cursor.
+      */
+     isNaked?: boolean;
+     /**
+      * Prop reserved for when component is wrapped by `<Link>` from react-router.
+      */
+     navigate?: () => void;
+     /**
+      * Callback when Button is pressed.
+      */
+     onClick?: (event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void;
+     /**
+      * Callback when focus leaves Button.
+      */
+     onBlur?: (event: FocusEvent<HTMLButtonElement | HTMLAnchorElement>) => void;
+     /**
+      * Callback when Button receives focus.
+      */
+     onFocus?: (event: FocusEvent<HTMLButtonElement | HTMLAnchorElement>) => void;
+     /**
+      * Specify the tabIndex of the button.
+      */
+     tabIndex?: number;
+     /**
+      * Useful when using button as an anchor tag.
+      */
+     target?: AnchorHTMLAttributes<HTMLAnchorElement>['target'];
+     /**
+      * The Button's type.
+      * NOTE: this is not restricted to button types since we allow
+      * rendering a button as a different HTML element than a button (`<a>` or `<input>`).
+      */
+     type?: 'submit' | 'reset' | 'button' | string;
+     /**
+      * The size of the button.
+      */
+     size?: ButtonSize | ResponsiveProp<ButtonSize>;
+     /**
+      * The color variant of the button
+      */
+     variant?: ButtonVariant;
+     /**
+      * ref - currently cannot be typed due to limitations of using the `as` prop
+      */
 }
 
-export const Button: FC<ButtonProps> = forwardRef(
+type AnchorButtonProps = { as: 'a'; } &
+  BaseButtonProps & Omit<JSX.IntrinsicElements['a'], 'ref'>
```

Comment:
Hey @dg-harris I saw your comment and understand these were perhaps problematic before, but I am not 100% following how this works. Won't omitting the ref attribute from JSX props type mean that passing a `ref` to the Button will now throw a TS error?

I'm sure I'm missing something but having some difficulty understanding this one.

Reply:
Ya no problem, it threw me off to. The issue is that the intrinsicElement refs are all `LegacyRef` but that's not compatible with `forwardRef`

The key here is that I've removed the explicit typing on `Button`  and added types to `forwardRef` so the `ref` prop actually comes from the `forwardRef` type signature.

The `FC` type actually doesn't allow refs so this was already a problem, it was just being masked by `[x: string]: any`